improver/spotdata/neighbour_finding.py

```diff
@@ -587,6 +587,12 @@ def process(self, sites, orography, land_mask):
         )
         data = np.expand_dims(data, 0).astype(np.float32)
 
+        # Sort the spot sites for better memory accessing later
+        for i in range(data.shape[0]):
+            site = data[i]
+            ind = np.lexsort((site[1], site[0]))
+            data[i] = site[:, ind]
+
```

Comment:
I think this sorting is changing the grid points associated with a site. It is fundamentally breaking the output, so needs to be thought about again.

This can be seen easily enough by looking at the acceptance test KGO that is produced by this new code and comparing it with the existing KGO.
```
old = iris.load('nearest_uk_kgo.nc') # old KGO
new = iris.load('nearest_uk_kgo.nc') # new KGO
# Choose some or other wmo_id, the example uses 3111
old_ex = old.extract(iris.Constraint(coord_values={'wmo_id':lambda cell: cell == '3111'}))
new_ex = new.extract(iris.Constraint(coord_values={'wmo_id':lambda cell: cell == '3111'}))
print(old_ex.data)
[[478.       550.        10.143832]]
print(new_ex.data)
[[502.       274.        10.143832]]
```
As you can see from this example, the indices have changed, so the data for this site will now be extracted from somewhere completely different and unrelated on the grid.